src/packagedcode/go_mod.py

```diff
@@ -73,140 +105,221 @@
-                gomod_data['namespace'] = parsed_module_name.group('namespace')
-                
-            parsed_require = cls.parse_require(line)
-            if parsed_require:
-                line_req = [parsed_require.group('namespace'), parsed_require.group('name'), parsed_require.group('version')]
-                require.append(line_req)
-
-            parsed_exclude = cls.parse_exclude(line)
-            if parsed_exclude:
-                line_exclude = [parsed_exclude.group('namespace'), parsed_exclude.group('name'), parsed_exclude.group('version')]
-                exclude.append(line_exclude)
-
-            if 'require' in line and '(' in line:
-                for req in lines[i+1:]:
-                    req = cls.preprocess(req)
-                    if ')' in req:
-                        break
-                    parsed_dep_link = cls.parse_dep_link(req)
-                    if parsed_dep_link:
-                        line_req = [parsed_dep_link.group('namespace'), parsed_dep_link.group('name'), parsed_dep_link.group('version')]
-                        require.append(line_req)
-
-            if 'exclude' in line and '(' in line:
-                for exc in lines[i+1:]:
-                    exc = cls.preprocess(exc)
-                    if ')' in exc:
-                        break
-                    parsed_dep_link = cls.parse_dep_link(exc)
-                    if parsed_dep_link:
-                        line_exclude = [parsed_dep_link.group('namespace'), parsed_dep_link.group('name'), parsed_dep_link.group('version')]
-                        exclude.append(line_exclude)
-
-        gomod_data['require'] = require
-        gomod_data['exclude'] = exclude
-
-        return gomod_data
+# Regex expressions to parse different types of go.sum file dependency
+# dep_type1 example: github.com/BurntSushi/toml v0.3.1 h1:WXkYY....
+parse_dep_type1 = re.compile(
+    r'(?P<namespace>(.*))'
+    r'\/'
+    r'(?P<name>[^\s]*)'
+    r'(\s)*'
+    r'(?P<version>(.*))'
+    r'/go.mod(\s)*h1:'
+    r'(?P<checksum>(.*))'
+).match
+
+# dep_type2 example: github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCN....
+parse_dep_type2 = re.compile(
+    r'(?P<namespace>(.*))'
+    r'/'
+    r'(?P<name>[^\s]*)'
+    r'(\s)*'
+    r'(?P<version>(.*))'
+    r'(\s)*h1:'
+    r'(?P<checksum>(.*))'
+).match
+
+
+def parse_gosum(location):
+    """
+    Return a list of GoSum from parsing the go.sum file at `location`.
+    """
+    with io.open(location, encoding='utf-8', closefd=True) as data:
+        lines = data.readlines()
+
+    gosums = []
+
+    for line in lines:
+        parsed_dep = parse_dep_type1(line)
+        if not parsed_dep:
+            parsed_dep = parse_dep_type2(line)
+
+        dep_obj = GoModule(
+                namespace=parsed_dep.group('namespace').strip(),
+                name=parsed_dep.group('name').strip(),
+                version=parsed_dep.group('version').strip()
+            )
+
+        if dep_obj not in gosums:
```

Comment:
Is this a possible case? or is this a side effect you treating lines with and without go.mod the same way?
If the later is true, then using the new proposed regex above and a test such as this may be more explicit. You can also then just yield too, rather than accumulate in a list
```
if parsed_dep.group('is_gomod'):
    continue
```

Reply:
Since we are not using checksums in the dependency object, so maximum time duplicate values will be present.